### PR TITLE
Add --set-opamp-endpoint to otelcol-config

### PR DESCRIPTION
**Pull request by echlebek**
This commit adds the set-opamp-endpoint action to otelcol-config. The set-opamp-endpoint action sets the OpAmp endpoint in sumologic-remote.yaml. If sumologic-remote.yaml does not exist, then an error is printed. The action also validates that the URL being set has a valid websocket scheme.

### CODE REVIEW

File: pkg/tools/otelcol-config/opamp.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	if err := validateEndpoint(endpoint); err != nil {
+		return err
+	}
+	conf, err := ReadConfigDir(ctx.ConfigDir)
+	if err != nil {
+		return err
+	}
+
+	writer := ctx.WriteSumologicRemote
+	doc := conf.SumologicRemote
```
Comment (ccressent) [inline pkg/tools/otelcol-config/opamp.go:24]: Nitpick: not 100% sure about the name of `doc`

File: pkg/tools/otelcol-config/opamp.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	encoder := yqlib.YamlFormat.EncoderFactory()
+	decoder := yqlib.YamlFormat.DecoderFactory()
+	eval := yqlib.NewStringEvaluator()
+
+	expression := fmt.Sprintf(".extensions.opamp.endpoint = %q", endpoint)
+	result, err := eval.EvaluateAll(expression, string(doc), encoder, decoder)
+	if err != nil {
+		return fmt.Errorf("couldn't write installation token: %s", err)
+	}
```
Comment (ccressent) [inline pkg/tools/otelcol-config/opamp.go:38]: Are you sure this is the correct error message?

**Review by ccressent — COMMENTED** [review]
Looking good; just see my comment about what I think is the wrong error message.

File: pkg/tools/otelcol-config/opamp.go (hunk 1 above)
Comment (echlebek) [reply pkg/tools/otelcol-config/opamp.go:24]: It's a yaml doc

File: pkg/tools/otelcol-config/opamp.go (hunk 2 above)
Comment (echlebek) [reply pkg/tools/otelcol-config/opamp.go:38]: Nope, this is incorrect, thanks!

**Review by ccressent — APPROVED** [review]

**Pull request merged**